File: java/src/main/java/it/giannibombelli/wsc2026/payment/domain/payment/PaymentStatus.java
package it.giannibombelli.wsc2026.payment.domain.payment;

public enum PaymentStatus {
    REQUESTED("requested"),
    PROCESSING("processing"),
    ACCEPTED("accepted"),
    REJECTED("rejected"),
    EXPIRED("expired"),
    REFUNDED("refunded");

    private final String value;

    PaymentStatus(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }
}

File: typescript/src/payment/domain/payment/paymentStatus.ts
import { requireArgument } from '@/common/utils/requireArgument.js';
// Stati dell'aggregato Payment.

export const PaymentStatus = {
  REQUESTED: 'REQUESTED',
  PROCESSING: 'PROCESSING',
  ACCEPTED: 'ACCEPTED',
  REJECTED: 'REJECTED',
  EXPIRED: 'EXPIRED',
  REFUNDED: 'REFUNDED',
} as const;

export type PaymentStatus = (typeof PaymentStatus)[keyof typeof PaymentStatus];

export function paymentStatusFromLabel(label: string): PaymentStatus {
  requireArgument(label, 'payment status label');
  const value = Object.values(PaymentStatus).find((status) => status === label);
  if (value === undefined) {
    throw new Error(`unknown payment status: ${label}`);
  }
  return value;
}
